quesma/clickhouse/table_discovery.go

```diff
@@ -184,11 +192,11 @@ func (td *tableDiscovery) readVirtualTables(configuredTables map[string]discover
 		}
 		discoTable := discoveredTable{
 			name:        virtualTable,
-			columnTypes: make(map[string]string),
+			columnTypes: make(map[string]columnMetadata),
 		}
 
 		for _, col := range readVirtualTable.Columns {
-			discoTable.columnTypes[col.Name] = col.Type
+			discoTable.columnTypes[col.Name] = columnMetadata{colType: col.Type}
```

Comment:
Do field encodings work for single table case? Or is it something left to do?

Reply:
This is not implemented yet